src/main.rs

```diff
@@ -1,3 +1,26 @@
-fn main() {
-    println!("ello");
+
+use gtree;
+use std::fs;
+use std::fs::File;
+use gitignore;
+use std::path::Path;
+
+fn main() -> std::io::Result<()> {
+    let maybe_file = File::open(".gitignore");
+    match maybe_file {
```

Comment:
It looks like this `File::open` and match is purely to check for the existence of the `.gitignore`, but you haven't actually dropped and closed the file so you end up with two handles to it I think.  At the very least you actually open the `File` twice. This is fine but suboptimal.

What you can do instead is create the `Path` as below and then use [`Path::is_file`](https://doc.rust-lang.org/std/path/struct.Path.html#method.is_file) to check for existence, then just pass the path directly to `gitignore::File::new`.

You could also not check for the existence of the file beforehand at all, and just use the `Result` returned from `gitignore::File::new`, and transform it into an `Option` with `Result::ok` and pass that directly into `print_gtree`.